appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/ResourceState.java

```diff
@@ -37,13 +37,15 @@
  * only if the new code is made subject to such option by the copyright
  * holder.
  */
+// Portions Copyright [2019] Payara Foundation and/or affiliates
 
 package com.sun.enterprise.resource;
 
 public class ResourceState {
     private boolean enlisted;
     private boolean busy;
     private long timestamp;
+    private TwiceBusyException busyException;
```

Comment:
I don't know much of the context here but race condition sounds like multi-threading which suggests there should thread safe handling somewhere here but I can't see any atomics, `volatile` or `synchronized`. There probably should be?

Reply:
Any comment on my question?

Reply:
No because the `TwiceBusyException` is used to store details of what has happened to help with tracing where the race condition occurs, there may be a deeper root cause of the issue, but the customer is happy with this.

Reply:
So the thread reading and writing the `ResourceState` is always the same?

Reply:
I think so